src/senaite/ast/view.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of SENAITE.AST.
+#
+# SENAITE.AST is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free
+# Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright 2020-2022 by it's authors.
+# Some rights reserved, see README and LICENSE.
+
+from bika.lims import _
+from bika.lims import api
+from Products.CMFCore.utils import getToolByName
+from Products.CMFPlone.utils import safe_unicode
+from Products.Five import BrowserView
+from senaite.ast.utils import copy_service
+from transaction import savepoint
+
+
+class DuplicateView(BrowserView):
+    """Displays a list of object in a table to duplicate.
+    """
+    created = []
+
+    def __call__(self):
+        uc = getToolByName(self.context, "uid_catalog")
+        if "copy_form_submitted" not in self.request:
+            uids = self.request.form.get("uids", [])
+            self.services = []
+            for uid in uids:
+                proxies = uc(UID=uid)
+                if proxies:
+                    self.services.append(proxies[0].getObject())
+            return self.template()
+        else:

```

Comment:
This `else` is not required. Note the previous condition returns `self.template()`